Highway.h

```diff
@@ -3,14 +3,20 @@
 #include <vector>
 #include <algorithm>
 
+#include "HighwayPatrol.h"
+#include "Vehicle.h"
+
 struct Highway
 {
     void changeSpeed(int newSpeed);
     void addVehicle(Vehicle* v);
     void removeVehicle(Vehicle* v);
-private:
+protected:
```

Comment:
Nothing is inheriting from `Highway`, so you don't need to change this access specifier from `private` to `protected.
the `HighwayPatrol` can access these functions when you make it a `friend`, which you have done.
- [ ] revise